### PR TITLE
feat: Add devenv

### PR DESCRIPTION
**Pull request by fschmtt**
### Description

Add [devenv](https://devenv.sh/) support to enable developers using devenv to easily get started:
* No manual installation of dependencies.
* Reproducible environments for all developers.
* Environment configured properly OOTB for running the *Getting started* section.

### Type of change

New feature

### ToDo's

- [ ] Check versions of installed packages (e.g. pnpm, node etc.)

### CODE REVIEW

File: devenv.nix [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+{
+  # Packages
+  packages = [
+    pkgs.nodePackages.pnpm
+  ];
+
+  # Languages
+  languages.javascript.enable = lib.mkDefault true;
+  languages.javascript.package = lib.mkDefault pkgs.nodejs_18;
```
Comment (fschmtt) [inline devenv.nix:11]: We can pin the node (major) version - is 18 correct?

File: devenv.nix [hunk 2]
```diff
@@ -0,0 +1,16 @@
+{ pkgs, lib, ... }:
+
+{
+  # Packages
+  packages = [
+    pkgs.nodePackages.pnpm
```
Comment (fschmtt) [inline devenv.nix:6]: This will use the latest version, is this OK?

File: devenv.nix [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+  ];
```
Comment (fschmtt) [inline devenv.nix:7]: Are additional packages required?

For me it was sufficient ot only add `pnpm`.

**Review by patzick — CHANGES_REQUESTED** [review]
Hey @fschmtt 
Please write more about your motives and what you are trying to achieve. Currently I see no reason for adding this to the monorepo.

File: devenv.nix (hunk 2 above)
Comment (shyim) [reply devenv.nix:6]: enable ```
languages.javascript.corepack.enable = true;
```

and remove pnpm as package. so it uses the pinned version in the package.json

**Comment by fschmtt** [comment]
Hey @patzick, for me as someone mainly using devenv for running all sorts of projects locally, it's just super nice to not have a "break" in DX 😄 

I updated the description of the PR acordingly.

I don't insist on merging this, if you feel like it's not contributing to the project, though 🙂

**Pull request closed without merging**